frontend/context/ServicesProvider.tsx

```diff
@@ -119,6 +119,8 @@ export const ServicesProvider = ({ children }: PropsWithChildren) => {
       (service) => service.service_config_id === selectedServiceConfigId,
     );
 
+    if (!selectedService) return;
```

Comment:
at some point we started returning this as an object (with `deploymentStatus`), so this stopped being null when the service is not created (right after onboarding) leading to this condition being not met and the "CannotStartAgentDueToUnexpectedError" being rendered

<img width="755" alt="image" src="https://github.com/user-attachments/assets/92173a8e-b0c4-4fae-94e3-ff5a4f879ad6">